core/src/main/java/org/infinispan/configuration/cache/SingleFileStoreConfigurationBuilder.java

```diff
@@ -103,6 +103,10 @@ public void validate() {
 
    @Override
    public void validate(GlobalConfiguration globalConfig) {
+      Attribute<String> location = attributes.attribute(LOCATION);
+      if (location.isNull() && !globalConfig.globalState().enabled()) {
+         Log.CONFIG.storeLocationRequired();
+      }
```

Comment:
We can remove this as the error will always be thrown on startup via `PersistenceUtil#getLocation`.

Reply:
Is it not preferable to validate invalid configuration when the configuration is built, rather than when a cache is started?

Reply:
Yes it is. The correct way is to validate this in the builders, I was trying to save some repetition but this approach makes more sense. The `SoftIndexFileStoreConfigurationBuilder` will need to be updated to check that the `DATA_LOCATION` and `INDEX_LOCATION` in their respective builders are not null. As well the `LOCATION` and `EXPIRATION_LOCATION` in the `RocksDBStoreConfigurationBuilder`.